### PR TITLE
Fix public maintenance lint findings

### PR DESCRIPTION
**Pull request by pderrenger**
Updates public maintenance lint handling for yolov5.\n\n- records intentional delayed-import exceptions for script entrypoints\n- resolves concrete Ruff findings in utility code\n- documents existing optional parameters surfaced by the sweep\n\nValidation:\n- ruff check --extend-select=F,I,D,UP,RUF,FA --target-version=py39 --ignore=D100,D104,D203,D205,D212,D213,D401,D406,D407,D413,RUF001,RUF002,RUF012 .\n- ruff format --check --line-length=120 train.py utils/augmentations.py utils/autoanchor.py utils/dataloaders.py utils/loss.py utils/metrics.py utils/segment/dataloaders.py utils/segment/loss.py utils/torch_utils.py\n- git diff --check\n- python -m compileall -q train.py utils/augmentations.py utils/autoanchor.py utils/dataloaders.py utils/loss.py utils/metrics.py utils/segment/dataloaders.py utils/segment/loss.py utils/torch_utils.py\n- PYTHONPATH=$PWD pytest tests/test_flask_rest_api.py tests/test_invariant_common.py -q

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
🧹 This PR is a code quality and maintainability cleanup for YOLOv5, improving linting support, clarifying documentation, and making several small readability-focused refactors without changing core behavior.

### 📊 Key Changes
- Added **Ruff per-file lint ignore rules** in `pyproject.toml` to better handle existing import-order and docstring exceptions across training, detection, segmentation, export, and test files.
- Improved **function documentation**:
  - `train.py`: clarified that `run(**kwargs)` accepts optional override arguments.
  - `utils/metrics.py`: documented the `eps` parameter in IoU-related helpers to explain its purpose.
- Refactored multiple code sections for **clearer variable names** and readability:
  - Renamed short variables like `l`, `s`, `m` to clearer names such as `label`, `segment`, `left`, and `right`.
  - Updated loops and comprehensions in augmentation, anchor checking, segmentation dataloaders, and loss-building code for better clarity.
- Modernized some Python syntax for **cleaner style**:
  - Replaced list concatenation patterns with unpacking, such as `[index, *random.choices(...)]`.
  - Replaced old `%` string formatting with **f-strings** in logging and model info output.
- Kept all changes focused on **code clarity and tooling compatibility**, with no major algorithmic or feature-level modifications.

### 🎯 Purpose & Impact
- **Makes the codebase easier to maintain** 🛠️ by improving naming, formatting, and inline documentation.
- **Helps developers use linting tools more smoothly** ✅, reducing noise from known exceptions while preserving code standards.
- **Improves readability for contributors** 👀, making training, augmentation, loss, and dataloader logic easier to follow.
- **Reduces confusion for users and maintainers** 📘 by clarifying API behavior and parameter meanings.
- **Low user-facing impact**: this PR is mainly an internal cleanup, so most users should not see behavior changes, but developers working on YOLOv5 should benefit from a cleaner and more consistent codebase.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @pderrenger, thank you for submitting a `ultralytics/yolov5` 🚀 PR! This is an automated message to help streamline review; an engineer will assist you soon. Please review the checklist below to help ensure a smooth integration. 😊

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/yolov5/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/yolov5` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: utils/torch_utils.py [hunk 1]
```diff
@@ -277,8 +277,7 @@ def model_info(model, verbose=False, imgsz=640):
         for i, (name, p) in enumerate(model.named_parameters()):
             name = name.replace("module_list.", "")
             print(
-                "%5g %40s %9s %12g %20s %10.3g %10.3g"
-                % (i, name, p.requires_grad, p.numel(), list(p.shape), p.mean(), p.std())
+                f"{i:5g} {name:>40s} {p.requires_grad!s:>9s} {p.numel():12g} {list(p.shape)!s:>20s} {p.mean():10.3g} {p.std():10.3g}"
```
Comment (UltralyticsAssistant) [inline utils/torch_utils.py]: ⚠️ **HIGH**: This f-string changes the formatting behavior in a way that can raise at runtime on the verbose `model_info()` path. In particular, `i` and `p.numel()` are integers, and the `g` presentation type is not valid for `int` with `str.format`/f-strings the way `%g` was here. `p.mean()` and `p.std()` are also safer to format as Python scalars explicitly. As written, `model_info(verbose=True)` can fail before printing any parameter stats.

**Suggested change:**
```suggestion
                f"{i:5d} {name:>40s} {p.requires_grad!s:>9s} {p.numel():12d} {list(p.shape)!s:>20s} {p.mean().item():10.3g} {p.std().item():10.3g}"
```

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Mostly a clean lint-focused PR with no obvious behavioral changes in most files. One regression stands out: the new f-string in `utils/torch_utils.py` can break `model_info(verbose=True)` due to incompatible numeric format specifiers.

💬 Posted 1 inline comment

File: utils/torch_utils.py (hunk 1 above)
Comment (pderrenger) [reply utils/torch_utils.py]: Fixed in 37888e0b by formatting integer fields with integer specifiers and converting tensor stats to Python scalars. Also smoke-tested model_info(verbose=True) locally.

**Comment by UltralyticsAssistant** [comment]
Merged! 🎉

Thank you @pderrenger for this thoughtful cleanup, and @glenn-jocher for the contributions and review support. As Leonardo da Vinci said, *“Simplicity is the ultimate sophistication.”* That fits this PR well: clearer naming, improved documentation, cleaner formatting, and better linting support all make YOLOv5 easier to maintain and contribute to without changing core behavior.

Really appreciate the care put into improving readability and developer experience here.
